manifest.go

```diff
@@ -416,6 +416,23 @@ func fromRawPruneOptions(raw rawPruneOptions) (gps.PruneOptions, gps.PruneProjec
 	return rootOptions, pruneProjects
 }
 
+func toRawPruneOptions(options gps.PruneOptions) rawPruneOptions {
```

Comment:
Let's add a unit test. I realize we have integration tests that are tangentially testing this, but they don't exercise all of the flags.

Reply:
After this is updated to reflect the changes in #1219 , how can we clarify that this only works during the initial init? Unmarshalling project prune options is not possible without too much hacking...

Reply:
@ibrasho I am not sure I understand the question/problem?

Reply:
Apologies for not explaining thoroughly. 😅

I'm not sure if you remember the issue with unmarshalling prune project options [[1]](https://github.com/golang/dep/blob/prune-release-prep/manifest.go#L54) [[2]](https://github.com/golang/dep/blob/prune-release-prep/gps/prune.go#L18-L29) (since we have a mixed string/bools TOML table)?

In short, this function (while it's only used in init) can be used later and assumed to write the parsed manifest as-read. While in reality, it would at best ignore prune project options.

This might not be an issue now, but I want us to make sure that we don't forget this later and assume that this function can un-marshall the full prune options.

Reply:
Just an observation: this function is also called during `ensure -add`.

Reply:
if we have a way of encoding that failure mode as a panic, that would be great. I'm not sure offhand if it's possible, but something like "check if the map of project-specific options is non-empty,and if so, panic"

Reply:
yeah...the call during the new mode is troubling. really, that should only be operating on discrete, additional items, not a whole `Manifest`.

Reply:
@ibrasho Let me repeat back what I am hearing to make sure I've got it.

The concern is that this function only writes global prune options. If later someone tried to read in the Gopkg.toml, modify the in-memory manifest, and then write back the entire manifest to disk that the project level prune options would be lost?

I read through the code for `dep ensure --add` and have seen the manifest output from this PR. It doesn't appear to be behaving incorrectly. So I'd like to set that aside for now.

If you agree that I've finally understood your concern(?), then my initial preference would have been to fully implement the conversion from a manifest to toml. Unfortunately, it appears from your comment that the conversion of project-level prune options to a raw representation is not easy and would require hacks. I'll take your word on that! 😀 

Since the full implementation is not actually needed, Sam's suggestion is to raise a panic when project-level prune options are present, forcing the developer to deal with this in the future if they try to use the unimplemented functionality.

Let me know if I've summarized correctly. If I have, then yes, I would also prefer a panic (and perhaps a copy/paste of my summary or a link back to it), over just a comment.

Reply:
A panic is a better approach.